sound/soc/sdca/sdca_asoc.c

```diff
@@ -954,8 +1072,13 @@ static int populate_control(struct device *dev,
 	(*kctl)->private_value = (unsigned long)mc;
 	(*kctl)->iface = SNDRV_CTL_ELEM_IFACE_MIXER;
 	(*kctl)->info = snd_soc_info_volsw;
-	(*kctl)->get = snd_soc_get_volsw;
-	(*kctl)->put = snd_soc_put_volsw;
+	if (strstr(control_name, "Volume")) {
```

Comment:
This should be done based of control->type instead of looking for strings in the name.

Reply:
Thanks. Will fix.